File: InsertionSort/CPP/InsertionSort.cpp
//
// Created by Shahriar Nasim Nafi on 28/6/22.
//

void insertionSort(int *array, int size) {
    for (int i = 1; i < size; i++) {
        int value = array[i];
        int j = i - 1;
        while (j >= 0 && value < array[j]) {
            array[j + 1] = array[j];
            j -= 1;
        }
        array[j + 1] = value;
    }
}

File: InsertionSort/Swift/InsertionSort.swift
//
// Created by Shahriar Nasim Nafi on 28/6/22.
//

extension Array where Element: Comparable {
    func insertionSort() -> [Element] {
        var sorted = self
        for i in 1..<sorted.count {
            let value = sorted[i]
            var j = i - 1
            while(j >= 0 && value < sorted[j]) {
                sorted[j + 1] = sorted[j]
                j-=1
            }
            sorted[j + 1] = value
        }
        return sorted
    }
}
